vasil-testnet/squash-fuel-utxo.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+#
+# Squash all fuel utxo of a signing key into a single output and mark it as fuel.
+
+set -e
+
+magic=9 # vasil testnet
+# magic=1097911063 # testnet
+
+sk=$1
+[[ -n $sk ]] && [[ -f $sk ]] || (echo "Missing argument: secret key file" && exit 1)
+
+vk=${sk%.sk}.vk
+if [[ ! -f "${vk}" ]]; then
+  cardano-cli key verification-key --signing-key-file ${sk} --verification-key-file ${vk}
+fi
+
+addr=$(cardano-cli address build --testnet-magic ${magic} --payment-verification-key-file $vk)
+
+fuelMarker="a654fb60d21c1fed48db2c320aa6df9737ec0204c0ba53b9b94a09fb40e757f3"
+utxo=$(cardano-cli query utxo \
+    --cardano-mode --epoch-slots 21600 \
+    --testnet-magic ${magic} \
+    --address $addr \
+    --out-file /dev/stdout | jq ". | select(.[].datumhash == \"${fuelMarker}\")")
+inputs=($(echo ${utxo} | jq -r '. | keys | @sh' | tr -d "'"))
+totalLovelace=$(echo ${utxo} | jq -r 'reduce .[] as $item (0; . + $item.value.lovelace)')
+
+echo "Squashing utxos (total: ${totalLovelace} lovelace):"
+txInArgs=""
+for input in ${inputs[@]}; do
+    echo " - ${input}"
+    txInArgs="${txInArgs}--tx-in ${input} "
+done
+
+estimatedFee=$(cardano-cli transaction build \
+    --babbage-era \
+    --cardano-mode --epoch-slots 21600 \
+    --testnet-magic ${magic} \
+    --script-valid \
+    ${txInArgs} \
+    --tx-out $addr+2000000 \
```

Comment:
why addr+2000000?
is this another magic number ;)

Reply:
It's 2 ADA, big enough to be `> minUTxOPerByte * txoutBytes` but required to estimate transaction fees with the `cardano-cli` using this way. It's a bit of a work-around because the `calculate-min-fee` command of `cardano-cli` is a bit annoying to use.